testing/test_config.py

```diff
@@ -352,6 +352,25 @@ def test_inifilename(self, tmpdir):
         assert config.inicfg.get('name') == 'value'
         assert config.inicfg.get('should_not_be_set') is None
 
+    @pytest.mark.issue1618
+    def test_consider_plugin(self, testdir):
+        pytest.importorskip('xdist')
```

Comment:
> i am certain that xdist is currently the only thing triggering the behavior

Something is wrong then... just tested on a fresh environment, without any plugins.

**master**

```
λ py.test test_foo.py -p no:plugin
Traceback (most recent call last):
...
  File "C:\Users\bruno\addopts-order\src\plugin.py", line 1, in <module>
    raise ValueError
ValueError
```

**RonnyPfannschmidt:fix-1618**
```
λ py.test test_foo.py -p no:plugin
============================= test session starts =============================
platform win32 -- Python 2.7.11, pytest-2.10.0.dev1, py-1.4.31, pluggy-0.3.1
rootdir: C:\Users\bruno\addopts-order\src, inifile:
collected 1 items

test_foo.py .

========================== 1 passed in 0.01 seconds =========================
```

That was on Python 2.7.


> the more interesting question is - why do these fail on ci, it works in my local test :/ i wonder where my environment is different

Should be `res.ret` instead of `res.res`, not sure why that passes for you.

I also executed your test on my new environment, and it passes if I don't have `xdist` and don't pass `-n1` (also changing to `res.ret`):

```python

import pytest

pytest_plugins = ['pytester']

def test_consider_plugin(testdir):
#    pytest.importorskip('xdist')
    print (testdir)
    testdir.makepyfile(conftest="""
            pytest_plugins = ['plugin']
        """,
        plugin="""
            raise ImportError
        """,
        test_foo="""
            def test():
                pass
        """,
    )
    res = testdir.inline_run(
        #'-n1',
        '-p', 'no:plugin')
    assert res.ret == 0
```

```
λ py.test test_ronny.py
============================= test session starts =============================
platform win32 -- Python 2.7.11, pytest-2.10.0.dev1, py-1.4.31, pluggy-0.3.1
rootdir: C:\Users\bruno\addopts-order\src, inifile:
collected 1 items

test_ronny.py .

========================== 1 passed in 0.10 seconds ===========================
```

The same test fails on **master** as expected:

```
ImportError: Error importing plugin "plugin"
```

So I'm guessing there is something weird with your local environment.